lms/djangoapps/discussion/rest_api/api.py

```diff
@@ -247,6 +253,12 @@ def _format_datetime(dt):
         return dt.isoformat().replace('+00:00', 'Z')
 
     course = _get_course(course_key, request.user)
+    user_roles = set(
+        Role.objects.filter(
+            users=request.user,
+            course_id=course.id,
+        ).values_list('name', flat=True).distinct()
+    )
```

Comment:
this should be moved to `edx-platform/lms/djangoapps/discussion/django_comment_client/utils.py` as `get_role_names`

Reply:
I've kept the name as `get_user_role_names` to clarify that it's returning user-specific role names and not all role names.